src/components/TextLink.tsx

```diff
@@ -0,0 +1,70 @@
+import React, {ForwardedRef, forwardRef, KeyboardEventHandler, MouseEventHandler, ReactElement} from 'react';
+import {GestureResponderEvent, Text as RNText, TextStyle} from 'react-native';
+import styles from '@styles/styles';
+import * as Link from '@userActions/Link';
+import CONST from '@src/CONST';
+import Text from './Text';
+
+type TextLinkProps = {
+    /** Link to open in new tab */
+    href: string;
```

Comment:
href is optional

```suggestion
    href?: string;
```

Reply:
Should it be optional?

Reply:
For me it makes more sense to have href required for a LinkText component, wdyt?

Reply:
i.e. FormAlertWrapper doesn't have href

Reply:
Looking at the code one of 'onPress' and 'href' should be defined:
```ts
    const openLink = (event) => {
        event.preventDefault();
        if (props.onPress) {
            props.onPress();
            return;
        }

        Link.openExternalLink(props.href);
    };
```

My proposal is to use discriminating union @situchan @MaciejSWM:
```ts
{ onPress: type; } | { href: type; }
```

Reply:
Implemented. I tested and it throws an error as expected when both `href` and `onPress` are defined. Nice @blazejkustra